### PR TITLE
Faults Debug - Fixing issues

### LINKED ISSUE #64: Fault Ui
1) Makes 3 different boxes to display faults in Motor 1-2 and Battery, as well as a scrollbar to showcase enough information.
2) Make a button for show history, when clicked would appear a pop up showing all the fault history of the car

### PR DESCRIPTION
**Pull request by badeniran**
Other faults debug pr went to crap after merging issues
Fixes #64

### CODE REVIEW

**Comment by badeniran** [comment]
Fixes #64

File: src/ViewLayer/DebugDisplay/ControlPage/ControlView/ControlView.cpp [hunk 1]
```diff
@@ -31,15 +31,15 @@ void ControlView::connectDriverControls(DriverControlsPresenter& driverControlsP
     connect(&driverControlsPresenter, SIGNAL(aliveReceived(bool)),
             this, SLOT(aliveReceived(bool)));
 
-    connect(&driverControlsPresenter, SIGNAL(headlightsLowReceived(bool)),
+    connect(&driverControlsPresenter, SIGNAL(lowHeadlightsReceived(bool)),
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/ControlPage/ControlView/ControlView.cpp]: What's with all the changes to this file?

File: src/ViewLayer/DebugDisplay/ControlPage/ControlView/ControlView.cpp.autosave [hunk 2]
```diff
@@ -0,0 +1,321 @@
+
+#include "ControlView.h"
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/ControlPage/ControlView/ControlView.cpp.autosave]: `.autosave` file. Delete

File: src/ViewLayer/DebugDisplay/MotorFaultPage/MotorFaultView/MotorFaultView.cpp [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+    , dclReducedDueToHighCellResistance_ ("DCL Reduced Due To High Cell Resistance")
+    , dclReducedDueToLowCellVoltage_ ("DCL Reduced Due To Low Cell Voltage")
+    , dclReducedDueToLowPackVoltage_ ("DCL Reduced Due To Low Pack Voltage")
+    , dclReducedDueToLowSoc_ ("DCL Reduced Due To Low SOC")
+    , dclReducedDueToTemperature_ ("DCL Reduced Due To Temperature")
+    , labelBCount_ (0)
+{
+
+    // Setting up Vertical bar
+    QScrollBar* verticalBar0 = new QScrollBar();
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/MotorFaultPage/MotorFaultView/MotorFaultView.cpp]: Rename this to `verticalBarM0`, same with `veritcalBar1`.

File: src/ViewLayer/DebugDisplay/MotorFaultPage/MotorFaultView/MotorFaultView.cpp [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+
+    verticalBar0->setStyleSheet(SCROLLBAR_STYLESHEET);
+    verticalBar1->setStyleSheet(SCROLLBAR_STYLESHEET);
+    verticalBarB->setStyleSheet(SCROLLBAR_STYLESHEET);
+
+    ui_.motor0ScrollArea().setVerticalScrollBar(verticalBar0);
+    ui_.motor1ScrollArea().setVerticalScrollBar(verticalBar1);
+    ui_.batteryScrollArea().setVerticalScrollBar(verticalBarB);
+
+    QLayout* layout0 = ui_.motor0ContentsWidget().layout();
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/MotorFaultPage/MotorFaultView/MotorFaultView.cpp]: Rename to `layoutM0`, same as `layoutM1`

File: src/ViewLayer/DebugDisplay/MotorFaultPage/MotorFaultView/MotorFaultView.cpp [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+    initializeLabel(dclReducedDueToTemperature_, layoutB, LIMIT_STYLESHEET);
+
+    ui_.motor0ContentsWidget().setLayout(layout0);
+    ui_.motor1ContentsWidget().setLayout(layout1);
+    ui_.batteryContentsWidget().setLayout(layoutB);
+
+    connectMotorFaults(motorFaultsPresenter_);
+    connectBatteryFaults(batteryFaultsPresenter_);
+
+    ErrorFlags teste;
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/MotorFaultPage/MotorFaultView/MotorFaultView.cpp]: Your testes are still here

File: src/ViewLayer/DisplayDashboard/DisplayDashboardView/DisplayDashboardView.cpp [hunk 6]
```diff
@@ -78,14 +78,14 @@ void DisplayDashboardView::connectDriverControls(DriverControlsPresenter& driver
...
 {
-    connect(&keyMotorPresenter, SIGNAL(motorSetCurrentReceived(double)),
-            this, SLOT(motorSetCurrentReceived(double)));
-    connect(&keyMotorPresenter, SIGNAL(motorActualSpeedReceived(double)),
-            this, SLOT(motorActualSpeedReceived(double)));
-    connect(&keyMotorPresenter, SIGNAL(motorBusVoltageReceived(double)),
-            this, SLOT(motorBusVoltageReceived(double)));
-    connect(&keyMotorPresenter, SIGNAL(motorBusCurrentReceived(double)),
-            this, SLOT(motorBusCurrentReceived(double)));
+    connect(&keyMotorPresenter, SIGNAL(motorSetCurrentReceived(double setCurrent)),
```
Comment (bill-luu) [inline src/ViewLayer/DisplayDashboard/DisplayDashboardView/DisplayDashboardView.cpp]: Why did this change?

File: src/ViewLayer/DebugDisplay/MotorFaultPage/MotorFaultView/MotorFaultView.cpp [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+    ui_.motor0ScrollArea().setVerticalScrollBar(verticalBar0);
+    ui_.motor1ScrollArea().setVerticalScrollBar(verticalBar1);
+    ui_.batteryScrollArea().setVerticalScrollBar(verticalBarB);
+
+    QLayout* layout0 = ui_.motor0ContentsWidget().layout();
+    QLayout* layout1 = ui_.motor1ContentsWidget().layout();
+    QLayout* layoutB = ui_.batteryContentsWidget().layout();
+
+    // Motor 0
+    initializeLabel(badMotorPositionHallSequence0Fault_, layout0, ERROR_STYLESHEET);
```
Comment (bill-luu) [inline src/ViewLayer/DebugDisplay/MotorFaultPage/MotorFaultView/MotorFaultView.cpp]: Now that I think more about it, it looks kinda cancer to have 100 lines of the same function in the constructor. Encapsulate all of this in an `initializeLabels()` function.

File: src/ViewLayer/ViewLayer.pro [hunk 8]
```diff
@@ -28,35 +31,33 @@ HEADERS += \
...
     DebugDisplay/MotorFaultPage/MotorFaultUi/MotorFaultUi.cpp \
     DebugDisplay/OverlordWidget/OverlordWidget.cpp \
     DebugDisplay/MotorPage/MotorUi/MotorUi.cpp \
     DebugDisplay/Tab/TabUi/TabUi.cpp \
     DebugDisplay/MPPTPage/MPPTUi/MpptUi.cpp \
     DebugDisplay/ControlPage/ControlView/ControlView.cpp \
-    DebugDisplay/MPPTPage/MPPTView/MpptView.cpp \
-    RaceModeDisplay/RaceModeDashboardView.cpp \
-    RaceModeDisplay/RaceModeDisplayUI/RaceModeDashboardUI.cpp
+    DebugDisplay/MPPTPage/MPPTView/MpptView.cpp
```
Comment (bill-luu) [inline src/ViewLayer/ViewLayer.pro]: Can you arrange these .pros in alphabetical order?

**Comment by bill-luu** [comment]
LGTM, feel free to merge once Lisa approves of the colors.

**Comment by badeniran** [comment]
Recent commit Fixes #90

**Comment by bill-luu** [comment]
Woops, closed but forgot to merge lol

**Pull request merged**